check-log/check-log.go

```diff
@@ -108,8 +110,18 @@ func validateMissing(missing string) bool {
 }
 
 func parseArgs(args []string) (*logOpts, error) {
+	var origArgs []string
+	copy(origArgs, args)
 	opts := &logOpts{}
 	_, err := flags.ParseArgs(opts, args)
+	opts.origArgs = origArgs
+	if opts.StateDir == "" {
+		workdir := os.Getenv("MACKEREL_PLUGIN_WORKDIR")
+		if workdir == "" {
+			workdir = os.TempDir()
+		}
+		opts.StateDir = filepath.Join(workdir, "check-log")
```

Comment:
[memo]
`StateDir` will be `MkdirAll`ed on `writeBytesToSkip` if necessary, so we don't have to `Mkdir` here